packages/jaeger-ui/src/utils/link-formatting.tsx

```diff
@@ -49,31 +49,66 @@ function getFormatFunctions<T = Trace[keyof Trace]>(): Record<
 
       return value.padStart(desiredLength, padCharacter);
     },
+
+    add: (value, offsetString: string) => {
+      if (typeof value !== 'number') {
+        console.error('add can only operate on numbers, ignoring formatting', {
+          value,
+          offsetString,
+        });
+        return value;
+      }
+
+      const offset = parseInt(offsetString, 10);
+      if (Number.isNaN(offset)) {
+        console.error('add needs a valid offset in microseconds as second argument, ignoring formatting', {
+          value,
+          offsetString,
+        });
+        return value;
+      }
+
+      return value + offset;
+    },
   };
 }
 
 export function getParameterAndFormatter<T = Trace[keyof Trace]>(
   parameter: string
 ): {
   parameterName: string;
-  formatFunction: ((value: T) => T | string) | null;
+  formatFunction: ((value: T) => T | string | number) | null;
 } {
   const parts = parameter.split('|').map(part => part.trim());
   const parameterName = parts[0];
   if (parts.length === 1) return { parameterName, formatFunction: null };
 
-  const [formatFunctionName, ...args] = parts[1].split(' ');
-
   const formatFunctions = getFormatFunctions<T>();
 
-  const formatFunction = formatFunctions[formatFunctionName];
-  if (!formatFunction) {
-    console.error('Unrecognized format function name, ignoring formatting', {
-      parameter,
-      formatFunctionName,
-      validValues: Object.keys(formatFunctions),
-    });
-  }
+  const chainedFormatFunctions = parts
+    .slice(1)
+    .map(formatPart => {
+      const [formatFunctionName, ...args] = formatPart.split(' ');
```

Comment:
what happens if there are multiple spaces?

Reply:
Good catch, handling this + adding a test case